src/streamlink/stream/dash_manifest.py

```diff
@@ -579,18 +740,21 @@ def segments(self, **kwargs):
         segmentTemplate = self.segmentTemplate or self.walk_back_get_attr("segmentTemplate")
 
         if segmentTemplate:
-            for segment in segmentTemplate.segments(RepresentationID=self.id,
-                                                    Bandwidth=int(self.bandwidth * 1000),
-                                                    **kwargs):
-                if segment.init:
-                    yield segment
-                else:
-                    yield segment
```

Comment:
This looks like this was done intentionally with future changes in mind in regards to the initialization segment. I still decided to replace it with `yield form` though...